src/Executor/Runner/Adapter.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace OpenRuntimes\Executor\Runner;
+
+use Swoole\Table;
+use Utopia\CLI\Console;
+use Utopia\DSN\DSN;
+use Utopia\Http\Http;
+use Utopia\Http\Response;
+use Utopia\Logger\Log;
+use Utopia\Storage\Device;
+use Utopia\Storage\Device\Backblaze;
+use Utopia\Storage\Device\DOSpaces;
+use Utopia\Storage\Device\Linode;
+use Utopia\Storage\Device\Local;
+use Utopia\Storage\Device\S3;
+use Utopia\Storage\Device\Wasabi;
+use Utopia\Storage\Storage;
+
+abstract class Adapter
+{
+    /**
+     * @param Table $activeRuntimes
+     * @param string[] $networks
+     * @param Table $statsHost
+     * @param Table $statsContainers
+     * @return void
+     */
+    abstract public function init(Table $activeRuntimes, array $networks, Table $statsHost, Table $statsContainers): void;
+
+    /**
+     * @param string $runtimeId
+     * @param int $timeout
+     * @param Response $response
+     * @param Log $log
+     * @return void
+     */
+    abstract public function getLogs(string $runtimeId, int $timeout, Response $response, Log $log): void;
+
+    /**
+     * @param string $runtimeId
+     * @param string $secret
+     * @param string $image
+     * @param string $entrypoint
+     * @param string $source
+     * @param string $destination
+     * @param string[] $variables
+     * @param string $runtimeEntrypoint
+     * @param string $command
+     * @param int $timeout
+     * @param bool $remove
+     * @param float $cpus
+     * @param int $memory
+     * @param string $version
+     * @param string $restartPolicy
+     * @param string[] $networks
+     * @param Table $activeRuntimes
+     * @param Log $log
+     * @return mixed
+     */
+    abstract public function createRuntime(string $runtimeId, string $secret, string $image, string $entrypoint, string $source, string $destination, array $variables, string $runtimeEntrypoint, string $command, int $timeout, bool $remove, float $cpus, int $memory, string $version, string $restartPolicy, array $networks, Table $activeRuntimes, Log $log): mixed;
+
+    /**
+     * @param string $runtimeId
+     * @param Table $activeRuntimes
+     * @param Log $log
+     * @return void
+     */
+    abstract public function deleteRuntime(string $runtimeId, Table $activeRuntimes, Log $log): void;
+
+    /**
+     * @param string $runtimeId
+     * @param string|null $payload
+     * @param string $path
+     * @param string $method
+     * @param mixed $headers
+     * @param int $timeout
+     * @param string $image
+     * @param string $source
+     * @param string $entrypoint
+     * @param mixed $variables
+     * @param float $cpus
+     * @param int $memory
+     * @param string $version
+     * @param string $runtimeEntrypoint
+     * @param bool $logging
+     * @param string $restartPolicy
+     * @param Table $activeRuntimes
+     * @param Log $log
+     * @return mixed
+     */
+    abstract public function createExecution(
```

Comment:
whats the difference between createRuntime and createExecution? Are they not the same?